### DISCUSSION TITLE
Would it be possible to support namespaced components?

### DISCUSSION THREAD

**Discussion by ahku** [unovue/reka-ui#356] [Ideas]
Since Vue supports namespaced components (`Select.Root` instead of `SelectRoot`), albeit only in script setup (I think?), wouldn't it be great to add support for that as well to align with the React version?

https://vuejs.org/api/sfc-script-setup.html#using-components

Great job so far!

**Comment by k11q** [comment]
Hi welcome! This is one of our earliest discussion, the resolution was basically we use a more vue-like convention(which is the current one) to allow people that started with vue to be familiar with it rather than prioritizing the ones with a radix background. If you still prefer it you can maybe create a community package that reexports the components with namespaced style. I hope this would answer your question

**Comment by zernonia** [comment]
I think this can be implemented via `unplugin-vue-components`, Will need to further look into this 😁

```ts
  // Allow subdirectories as namespace prefix for components.
  directoryAsNamespace: false,
```


the idea is to export a `radix-vue/resolver`, that would resolve the mentioned plugin, and user can defined how they want the components to be imported. eg `prefix`, or `namespaced`

I have created a ticket https://github.com/radix-vue/radix-vue/issues/362

**Reply by ahku** [reply]
Hi, I saw that the ticket was closed. Does this mean that the directoryAsNamespace was implemented, or that we found another solution?

**Comment by zernonia** [comment]
Decided to implement this https://github.com/radix-vue/radix-vue/issues/75 😁
